### PR TITLE
Ucrvariables as factor

### PR DESCRIPTION
**Pull request by Patowhiz**
@africanmathsinitiative/developers this solves issue Focus wrong when I return to a dialogue #4250 .
Changes made are to the UcrvariablesAsFactor control and all the other dialogues that use the control.
Its ready for review.

### CODE REVIEW

File: instat/ucrVariablesAsFactor.vb [hunk 1]
```diff
@@ -77,150 +77,152 @@ Public Class ucrVariablesAsFactor
...
-                ucrVariableSelector.ucrAvailableDataFrames.clsCurrDataFrame.RemoveParameterByName("measure.vars")
-                ucrVariableSelector.ucrAvailableDataFrames.clsCurrDataFrame.RemoveParameterByName("id.vars")
+        If bSingleVariable Then
+            'need to translate correctly
+            cmdVariables.Text = "Single Variable"
+            cmdVariables.FlatStyle = FlatStyle.Popup
+            ucrSingleVariable.Visible = True
+            ucrMultipleVariables.Visible = False
+            If ucrVariableSelector IsNot Nothing Then
+                'ucrSingleVariable.SetMeAsReceiver()
```
Comment (Patowhiz) [inline instat/ucrVariablesAsFactor.vb]: The change

**Comment by dannyparsons** [comment]
See my comment about a possible additional change needed here https://github.com/africanmathsinitiative/R-Instat/issues/4250#issuecomment-349635610
I haven't checked the code so I might be completely wrong on this, but worth testing before accepting this.

**Comment by maxwellfundi** [comment]
@dannyparsons  this works great. Its ready for your review. Additions done and looks good now.

File: instat/ucrVariablesAsFactor.vb (hunk 1 above)
Comment (dannyparsons) [reply instat/ucrVariablesAsFactor.vb]: Great looks good. You can remove these lines that you have commented out.

Comment (Patowhiz) [reply instat/ucrVariablesAsFactor.vb]: @dannyparsons these I have removed the comments

**Pull request merged**